aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -254,6 +254,30 @@
    */
   public static final String S3_WRITE_TAGS_PREFIX = "s3.write.tags.";
 
+  /**
+   * Used by {@link S3FileIO} to tag objects when deleting. When this config is set, objects are
+   * tagged with the configured key-value pairs instead of being hard-deleted. This is considered a
```

Comment:
based on the new behavior, the line here should say "objects are tagged with the configured key-value pairs before deletion"